product_sales_record/models/product_template.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons import decimal_precision as dp
+
+from odoo import models, fields, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    list_price_record = fields.Monetary(
+        string='Sale Price Record',
+        digits = dp.get_precision('Product Price'),
+        readonly=True,
+        store=True,
+    )
+    gross_profit = fields.Monetary(
+        string='Gross Profit',
+        digits=dp.get_precision('Product Price'),
+        compute='_compute_gross_profit',
+        readonly=True,
+        store=True,
+    )
+    confirmation_date = fields.Datetime(
+        string='Confirmation Date',
+        readonly=True,
+    )
+    team_id = fields.Many2one(
+        'crm.team',
+        string='Sales Channel',
+        readonly=True,
+    )
+
+    @api.depends('list_price_record', 'list_price', 'standard_price')
+    def _compute_gross_profit(self):
+        for product in self:
+            if product.standard_price:
+                if product.list_price_record and \
+                                product.list_price_record != 0:
+                    product.gross_profit = product.list_price_record - \
+                                           product.standard_price
+                elif product.list_price:
+                    product.gross_profit = product.list_price - \
+                                           product.standard_price
```

Comment:
We might want to change the design a bit.  A potential issue is that some products are actually sold for free (sale_price_unit = 0).  When this happens (not sure if such case exists), according to the current logic, the gross profit does not represent the actual profit (or loss in this case given standard_price is positive).

Therefore, we may reconstruct this part of the logic by using sale_count field of product.template.

```
    @api.depends('confirmation_date', 'sale_price_unit', 'list_price', 'standard_price')
    def _compute_gross_profit(self):
        for pt in self:
            if pt.sale_count:
                pt.gross_profit = pt.sale_price_unit - pt.standard_price
            else:
                pt.gross_profit = pt.list_price - pt.standard_price
```

Note that `'confirmation_date'` is added as a trigger as I wasn't sure if writing `0` to sale_price_unit would trigger this method in case a product is sold for free.  Otherwise, I think above code should work.  Can you check if this makes sense?

Reply:
The method can be triggered even writing 0 to `sale_price_unit`. I will proceed with the changes.